src/storageManager.js

```diff
@@ -10,31 +11,44 @@ export let storageCallbacks = [];
  * Storage options
  * @typedef {Object} storageOptions
  * @property {Number=} gvlid - Vendor id
- * @property {string} moduleName - Module name
+ * @property {string} moduleName? - Module name
+ * @property {string=} bidderCode? - Bidder code
  * @property {string=} moduleType - Module type, value can be anyone of core or prebid-module
  */
 
 /**
  * Returns list of storage related functions with gvlid, module name and module type in its scope.
  * All three argument are optional here. Below shows the usage of of these
  * - GVL Id: Pass GVL id if you are a vendor
- * - Module name: All modules need to pass module name
+ * - Bidder code: All bid adapters need to pass bidderCode
+ * - Module name: All other modules need to pass module name
  * - Module type: Some modules may need these functions but are not vendor. e.g prebid core files in src and modules like currency.
  * @param {storageOptions} options
  */
-export function newStorageManager({gvlid, moduleName, moduleType} = {}) {
+export function newStorageManager({gvlid, moduleName, bidderCode, moduleType} = {}, {bidderSettings = defaultBidderSettings} = {}) {
+  function isBidderDisallowed() {
+    if (bidderCode == null) {
+      return false;
+    }
+    const storageAllowed = bidderSettings.get(bidderCode, 'storageAllowed');
```

Comment:
should this also try to use the moduleName as well?

When I try this:

```js
        pbjs.bidderSettings = {
            rubicon: {
                storageAllowed: false
            }
        };
```

And use the rubiconAnalyticsAdapter, I still see the module is allowed to set its storage:

![image](https://user-images.githubusercontent.com/13972794/154162919-88aee730-7f59-42a2-a616-9e2267074923.png)

Reply:
Also, is there a way to set this globally? as to deny storageAllowed for all in bidderSettings?

```js
        pbjs.bidderSettings = {
            standard: {
                storageAllowed: false
            },
        };
```

for example

Reply:
@robertrmartinez, my understanding is that `bidderSettings` should control bid adapters, not analytics adapters (which is where `rubicon` controls its cookie from). That was the motivation for adding the new `bidderCode` parameter to the storage layer. The `moduleName` is treated as it was before, it can disable storage but you'd have to use consent management.

and yes, `bidderSettings.standard` is the fallback  - the logic is in `bidderSettings.js` (https://github.com/prebid/Prebid.js/pull/7712)

Reply:
Ok, so even when I set 

```js
        pbjs.bidderSettings = {
            standard: {
                storageAllowed: false
            },
        };
```

It is not expected to stop Analytic Adapters from storing?

Ok 👍

Reply:
Though, I would say, if a pub just wants to stop storage across the board, it kind of sucks they have to use the consentManagement module, if they do not even want consentManagement installed.

But I'll leave that to the deciders!

Reply:
Updated https://github.com/prebid/prebid.github.io/pull/3540 to be more explicit on what this setting does.